src/goto-programs/remove_exceptions.cpp

```diff
@@ -287,6 +293,49 @@ void remove_exceptionst::instrument_exception_handler(
   instr_it->make_skip();
 }
 
+/// Find the innermost universal exception handler
+/// (with exception type any). We record this one
+/// because no handler after it can possibly catch.
+/// @param stack_catch exception table
+/// @param default_try returns the try block
+///        corresponding to the desired exception handler
+/// @param default_try returns the catch block
+///        corresponding to the exception desired exception handler
+/// @return the desired exception handler
+goto_programt::targett remove_exceptionst::findUniversalException(
+  const remove_exceptionst::stack_catcht &stack_catch,
+  goto_programt &goto_program,
+  std::size_t &default_try,
+  std::size_t &default_catch)
+{
+  // Unless we have a universal exception handler, jump to end of function
+  // if not caught:
+  goto_programt::targett default_target=goto_program.get_end_function();
+  for(std::size_t i=stack_catch.size(); i>=0; --i)
+  {
+    if(default_try!=0)
+      break;
+    for(std::size_t j=0; j<stack_catch[i].size(); ++j)
+    {
+      goto_programt::targett new_state_pc=
+        stack_catch[i][j].second;
+      if(stack_catch[i][j].first.empty())
+      {
+        // Universal handler. Highest on the stack takes
+        // precedence, so overwrite any we've already seen:
+        default_target=new_state_pc;
+
+        // Record the position of the default behaviour as any further catches
+        // will not capture the throw
+        default_try=i;
+        default_catch=j;
+        break;
```

Comment:
Here we always break out of both loops and return right? Maybe change this to a return then you can get rid of the if statement at the start of the outer loop?

Reply:
this only breaks the first loop

Reply:
But sets `default_try` to `i` which is either 0, in which case the loop is at it's conclusion, or not 0, in which case the if at the start of the outer loop will cause it to break no?